client/app/pages/queries-list/QueriesList.jsx

```diff
@@ -0,0 +1,205 @@
+import { currentUser } from '@/services/auth';
+import navigateTo from '@/services/navigateTo';
+import { routesToAngularRoutes } from '@/lib/utils';
+
+import QueriesListEmptyState from './QueriesListEmptyState';
+
+import './queries-list.css';
+
+class QueriesList extends React.Component {
+  static propTypes = {
+    currentPage: PropTypes.string.isRequired,
+  };
+
+  static routes = [
+    {
+      path: '/queries',
+      title: 'Queries',
+      key: 'all',
+    },
+    {
+      path: '/queries/favorites',
+      title: 'Favorite Queries',
+      key: 'favorites',
+    },
+    {
+      path: '/queries/archive',
+      title: 'Archived Queries',
+      key: 'archive',
+    },
+    {
+      path: '/queries/my',
+      title: 'My Queries',
+      key: 'my',
+    },
+  ];
+
+  static sidebarMenu = [
+    {
+      key: 'all',
+      href: 'queries',
+      title: 'All Queries',
+    },
+    {
+      key: 'favorites',
+      href: 'queries/favorites',
+      title: 'Favorites',
+      icon: () => <Sidebar.MenuIcon icon="fa fa-star" />,
+    },
+    {
+      key: 'archive',
+      href: 'queries/archive',
+      title: 'Archive',
+      icon: () => <Sidebar.MenuIcon icon="fa fa-archive" />,
+    },
+    {
+      key: 'my',
+      href: 'queries/my',
+      title: 'My Queries',
+      icon: () => <Sidebar.ProfileImage user={currentUser} />,
+      isAvailable: () => currentUser.hasPermission('create_query'),
+    },
+  ];
+
+  static listColumns = [
+    Columns.favorites({ className: 'p-r-0' }),
+    Columns.custom.sortable((text, item) => (
+      <React.Fragment>
+        <a className="table-main-title" href={'dashboard/' + item.slug}>{ item.name }</a>
+        <QueryTagsControl
+          className="d-block"
+          tags={item.tags}
+          isDraft={item.is_draft}
+          isArchived={item.is_archived}
+        />
+      </React.Fragment>
+    ), {
+      title: 'Name',
+      field: 'name',
+      width: null,
+    }),
+    Columns.avatar({ field: 'user', className: 'p-l-0 p-r-0' }, name => `Created by ${name}`),
+    Columns.dateTime.sortable({ title: 'Created At', field: 'created_at' }),
+    Columns.duration.sortable({ title: 'Runtime', field: 'runtime' }),
+    Columns.dateTime.sortable({ title: 'Last Executed At', field: 'retrieved_at', orderByField: 'executed_at' }),
+    Columns.custom.sortable(
+      (text, item) => <SchedulePhrase schedule={item.schedule} isNew={item.isNew()} />,
+      { title: 'Update Schedule', field: 'schedule' },
+    ),
+  ];
+
+  constructor(props) {
+    super(props);
+
+    const resources = {
+      all: Query.query.bind(Query),
+      my: Query.myQueries.bind(Query),
+      favorites: Query.favorites.bind(Query),
+      archive: Query.archive.bind(Query),
+    };
+    const resource = resources[this.props.currentPage];
+
+    this.controller = new LiveItemsList({
+      doRequest: LiveItemsList.createFetcher(resource, Query),
+      onChange: ({ state }) => this.setState(state),
```

Comment:
Giving control of the state to a single controller (not talking about politics here) seems non-ideal. What I like about hoc is that each "ability" comes with it's own set of props so you can add as much as you like - but with this implementation, an additional ability alongside `LiveItemsList` won't be possible. Wdyt?

Reply:
> One disadvantage about this is that we won't be able to add QueriesList specific state props, cause they'll get overwritten when live items change.

You can always set all data that came from controller as a single object. In this case I have no additional props, so decided to do it this way.

> What I like about hoc is that each "ability" comes with it's own set of props so you can add as much as you like

The common issue of all this pages is that you cannot just have own "controller" for, say, pagination, sorting, search, etc. - all that parts are coupled. Even if I'll implement it as HoC - it will be the single wrapper with all that logic. Even more - I tried to group state props somehow, but the only group I can definitely create - pagination; finally I decided just to put all state props into a plain object. I can consider refactoring `LiveItemsList` to HoC, but only as a single component - otherwise I'll put too much effort probably with no extra value from it.

Reply:
@ranbena Implemented `LiveItemsList` as HOC - really looks almost perfect :thinking: